src/AI/WallyAI.java

```diff
@@ -259,6 +260,8 @@ else if( unitQueue.isEmpty() )
           {
             for( Unit target : unitLists.get(co) )
             {
+              if (target.getHP() < 1) // Try not to pick fights with zombies
```

Comment:
Does this happen often? Or ever?

Reply:
It did. Caused an infinite loop.
It was due to the "AIs can sometimes do the same action twice in a row" bug, also fixed in this PR.
An Arty shot a Recon twice, the Recon sat at 0 HP, and Wally got into an infinite loop.